File: app/src/main/java/com/pengxh/autodingding/adapter/BaseFragmentAdapter.java
package com.pengxh.autodingding.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.List;

public class BaseFragmentAdapter extends FragmentPagerAdapter {

    private final List<Fragment> pageList;

    public BaseFragmentAdapter(@NonNull FragmentManager fm, List<Fragment> pageList) {
        super(fm);
        this.pageList = pageList;
    }

    @Override
    public Fragment getItem(int position) {
        return pageList.get(position);
    }

    @Override
    public int getCount() {
        return pageList.size();
    }
}

File: app/src/main/java/com/pengxh/autodingding/adapter/BaseFragmentAdapter.kt
package com.pengxh.autodingding.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class BaseFragmentAdapter(manager: FragmentManager, private val pages: List<Fragment>) :
    FragmentPagerAdapter(manager) {
    override fun getItem(position: Int): Fragment = pages[position]

    override fun getCount(): Int = pages.size
}